node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -506,48 +591,60 @@ func (o OsDeviceConnectivityHelperGeneric) ReloadMultipath() error {
 	return nil
 }
 
-func (o OsDeviceConnectivityHelperGeneric) GetDmsPath(volumeId string, volumeNguid string) (string, error) {
-	volumeUuidLower := strings.ToLower(volumeId)
-
-	mpathdOutput, err := o.Helper.WaitForDmToExist(volumeUuidLower, volumeNguid, WaitForMpathRetries, WaitForMpathWaitIntervalSec)
-
+func (o OsDeviceConnectivityHelperGeneric) GetMpathdOutputByVolumeId(volumeIdVariations []string) (string, error) {
+	mpathdOutput, err := o.Helper.GetMpathdOutput(volumeIdVariations, multipathdWildcardsMpathNameAndVolumeId)
 	if err != nil {
 		return "", err
 	}
+	return mpathdOutput, nil
+}
 
-	if mpathdOutput == "" {
-		return "", &MultipathDeviceNotFoundForVolumeError{volumeId}
+func (o OsDeviceConnectivityHelperGeneric) GetVolumeIdByMpathDeviceName(mpathDeviceName string, mpathdOutput string) (string, error) {
+	mpathDeviceNames := []string{mpathDeviceName}
+	volumeIds := o.Helper.ExtractDmFieldValues(mpathDeviceNames, mpathdOutput)
+	volumeId, err := getUniqueDmFieldValue(volumeIds, mpathDeviceName)
+	if err != nil {
+		return "", err
 	}
 
-	dms := make(map[string]bool)
-	scanner := bufio.NewScanner(strings.NewReader(mpathdOutput))
-	for scanner.Scan() {
-		deviceLine := scanner.Text()
-		lineParts := strings.Split(deviceLine, mpathdSeparator)
-		dm, uuid := lineParts[0], lineParts[1]
-		if strings.Contains(uuid, volumeUuidLower) || strings.Contains(uuid, volumeNguid) {
-			dmPath := filepath.Join(DevPath, filepath.Base(strings.TrimSpace(dm)))
-			dms[dmPath] = true
-			logger.Infof("GetMpathDevice: DM found: %s for volume %s", dmPath, uuid)
-		}
-	}
+	return volumeId, nil
+}
 
-	if len(dms) > 1 {
-		return "", &MultipleDmDevicesError{volumeId, dms}
+func (o OsDeviceConnectivityHelperGeneric) GetDmsPath(volumeIdVariations []string) (string, error) {
+	mpathdOutput, err := o.Helper.GetMpathdOutput(volumeIdVariations, MultipathdWildcardsMpathAndVolumeId)
+	if err != nil {
+		return "", err
 	}
+	dms := o.Helper.ExtractDmFieldValues(volumeIdVariations, mpathdOutput)
+	return o.Helper.GetFullDmPath(dms, volumeIdVariations[0])
+}
 
-	var dm string
-	for dm = range dms {
-		break // because its a single value in the map(1 mpath device, if not it should fail above), so just take the first
+func (OsDeviceConnectivityHelperGeneric) GetMpathDeviceNameFromProcMounts(procMounts string, volumePath string) (string, error) {
+	scanner := bufio.NewScanner(strings.NewReader(procMounts))
+	for scanner.Scan() {
+		mountLine := scanner.Text()
+		if strings.Contains(mountLine, volumePath) {
+			return getMpathDeviceNameFromMountLine(mountLine), nil
+		}
 	}
+	return "", &MultipathDeviceNotFoundForVolumePathError{volumePath}
+}
 
-	return dm, nil
+func getMpathDeviceNameFromMountLine(mountLine string) string {
+	lineParts := strings.Fields(mountLine)
+	fullMpathDeviceName := lineParts[0]
+	splitedFullMpathDeviceName := strings.Split(fullMpathDeviceName, "/")
+	return splitedFullMpathDeviceName[len(splitedFullMpathDeviceName)-1]
```

Comment:
`split(t)ed` is not a word.
```suggestion
	mpathDevicePath := lineParts[0]
	mpathDevicePathParts := strings.Split(mpathDevicePath, "/")
	return mpathDevicePathParts[len(mpathDevicePathParts)-1]
```
also consider [not parsing it yourself](https://stackoverflow.com/a/34527720)

Reply:
done